torch/_inductor/config.py

```diff
@@ -460,6 +460,12 @@ def prologue_fusion_enabled() -> bool:
     == "1"
 )
 
+# A tuple of customized partition wrappers from users. Inductor mechanically wraps
+# each partition fn with the customized partition wrapper. When using for CUDAGraph wrapper,
+# users need to handle details such as static inputs, dynamic shapes, etc.
+customized_partition_wrappers: Optional[
+    tuple[Callable[..., Callable[..., Any]], ...]
+] = None
```

Comment:
imo tuple is the wrong api. how do the users know how many to provide? Can we instead have an api that takes in a function and other metadata about the function - potentially partition index within function, graph index. 

It could be 

def wrap_cudagraph(fn, CudaGraphMeta)

so that we can add new fields without breaking bc.. 

Or at least mark this api as unstable..

Reply:
Thanks for the great suggestion! I updated the interface to be `def wrapper(fn: PartitionFnType, metadata: CUDAGraphWrapperMetadata) -> PartitionFnType` which is more flexible.